src/prefab_manager.zig

```diff
@@ -0,0 +1,197 @@
+        defer arena_state.deinit();
+        const arena = arena_state.allocator();
+
+        var prefab = world.newPrefab(path);
+        prefab.setOverride(fd.Position.init(0, 0, 0));
+        prefab.setOverride(fd.EulerRotation{});
+        prefab.setOverride(fd.Scale.createScalar(1));
+        prefab.setOverride(fd.Transform{});
+        prefab.setOverride(fd.Forward{});
+        prefab.setOverride(fd.Dynamic{});
+        self.parseNode(scene.nodes.?[0], prefab, world, gfxstate, arena);
+
+        self.prefab_hash_map.put(path_id, prefab) catch unreachable;
+
+        return prefab;
+    }
+
+    pub fn getPrefabByPath(self: *@This(), path: [:0]const u8) ?flecs.Entity {
+        const path_id = IdLocal.init(path);
+        var existing_prefab = self.prefab_hash_map.get(path_id);
+        if (existing_prefab) |prefab| {
+            return prefab;
+        }
+
+        return null;
+    }
+
+    pub fn instantiatePrefab(self: @This(), world: *flecs.World, prefab: flecs.Entity) flecs.Entity {
+        const entity = world.newEntity();
+        entity.addPair(self.is_a, prefab);
+        return entity;
+    }
+
+    fn parseNode(
+        self: *@This(),
+        node: *zcgltf.Node,
+        parent_entity: flecs.Entity,
+        world: *flecs.World,
+        gfxstate: *gfx.D3D12State,
+        arena: std.mem.Allocator,
+    ) void {
+        // Set parent
+        var entity = world.newPrefab(node.name);
+        entity.addPair(flecs.c.Constants.EcsChildOf, parent_entity);
+        entity.setOverride(fd.Forward{});
+        entity.setOverride(fd.Dynamic{});
+        entity.setOverride(fd.Transform{});
+
+        // Set transform
+        var position = fd.Position.init(0, 0, 0);
+        var rotation = fd.EulerRotation{};
+        var scale = fd.Scale.createScalar(1);
+
+        if (node.has_rotation != 0) {
+            var euler = zm.quatToRollPitchYaw(node.rotation);
+            rotation.roll = euler[0];
+            rotation.pitch = euler[1];
+            rotation.yaw = euler[2];
+        }
+
+        if (node.has_translation != 0) {
+            position.x = node.translation[0];
+            position.y = node.translation[1];
+            position.z = node.translation[2];
+        }
+
+        if (node.has_scale != 0) {
+            scale.x = node.scale[0];
+            scale.y = node.scale[1];
+            scale.z = node.scale[2];
+        }
+        entity.setOverride(position);
+        entity.setOverride(rotation);
+        entity.setOverride(scale);
+
+        // Parse and assign mesh
+        if (node.mesh != null) {
+            var static_mesh_component: fd.StaticMeshComponent = undefined;
+
+            // TODO(gmodarelli): Cycle through all primitives
+            assert(node.mesh.?.primitives_count == 1);
+            const primitive = &node.mesh.?.primitives[0];
+
+            const mesh_name = util.asConstSentinelTerminated(node.mesh.?.name.?);
+
+            if (gfxstate.findMeshByName(mesh_name)) |mesh_handle| {
```

Comment:
Does this mean we can currently have two different prefabs using the same mesh? (without having more mesh data) For example a spider boss that's the same but 3x scaled.

Reply:
Yes :)